src/main/java/digital/slovensko/autogram/core/XDCTransformer.java

```diff
@@ -213,4 +245,69 @@ private Element createUsedPresentationSchemaReference() {
     private static String toNamespacedString(DigestAlgorithm digestAlgorithm) {
         return "urn:oid:" + digestAlgorithm.getOid();
     }
+
+    public boolean validateXsdDigest() {
+        String xsdSchemaHash = computeDigest(xsdSchema);
+        String xsdDigestValue = getDigestValueFromElement("UsedXSDReference");
+        return xsdSchemaHash.equals(xsdDigestValue);
+    }
+
+    public boolean validateXsltDigest() {
+        String xsltSchemaHash = computeDigest(xsltSchema);
+        String xsltDigestValue = getDigestValueFromElement("UsedPresentationSchemaReference");
+        return xsltSchemaHash.equals(xsltDigestValue);
+    }
+
+    private String getDigestValueFromElement(String elementLocalName) {
+        var xdc = document.getDocumentElement();
+
+        var xmlData = xdc.getElementsByTagNameNS("http://data.gov.sk/def/container/xmldatacontainer+xml/1.1", elementLocalName)
+                .item(0);
+        if (xmlData == null)
+            throw new RuntimeException("XMLData not found in XDC");
+
+        var attributes = xmlData.getAttributes();
+        if (attributes == null)
+            throw new RuntimeException("Attributes not found");
+
+        var digestValue = attributes.getNamedItem("DigestValue");
+        if (digestValue == null)
+            throw new RuntimeException("DigestValue not found");
```

Comment:
Tieto RuntimeExceptions (aj všetky ďalšie v tejto classe) hádžu do response 500 - INTERNAL_ERROR, čo ale nie je pravda. Malo by to hádzať nejakú 422. Možno treba vytvoriť nejakú XDCException alebo použiť RequestValidationException.